### PR TITLE
Update Cloud Storage Docs

### PR DESCRIPTION
**Pull request by careless25**
**Proposed changes**: Documentation was lacking on how the storing and retrieval of model works.
Might help with: 
https://github.com/RasaHQ/rasa_nlu/issues/1426
- ...

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible

File: docs/persist.rst [hunk 1]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
 
     If there is no container with the name ``AZURE_CONTAINER`` Rasa will create it.
 
-Models are gzipped before saving to cloud.
+Models are gzipped before saving to cloud. The gzipped file naming convention
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
Models are gzipped before they are saved in the cloud. The gzipped file naming convention
```

File: docs/persist.rst [hunk 2]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+is `{PROJECT}___{MODEL_NAME}.tar.gz` and should be in the root folder of the storage service.
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
is `{PROJECT}{MODEL_NAME}.tar.gz` and it is stored in the root folder of the storage service.
```

File: docs/persist.rst [hunk 3]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+Currently, you are not able to specify the path on cloud storage.
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
Currently, you are not able to manually specify the path on cloud storage.
```

File: docs/persist.rst [hunk 4]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+
+If storing trained models, rasa_nlu will gzip the new model and upload to the root. If retrieving/loading models
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
If storing trained models, rasa_nlu will gzip the new model and upload to the container. If retrieving/loading models
```

File: docs/persist.rst [hunk 5]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+from the cloud storage, rasa_nlu will download the gzipped model locally an extract the contents to the location
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
from the cloud storage, Rasa NLU will download the gzipped model locally and extract the contents to the location
```

**Comment by wochinge** [comment]
@careless25 Thanks for your contribution. I left a few comments. Implement them and then the change is ready to go 🚀

File: docs/persist.rst (hunk 2 above)
Comment (careless25) [reply docs/persist.rst]: Was this behavior changed in v0.14? We are using v0.13.7 so I tested on that.

Comment (careless25) [reply docs/persist.rst]: Line 96-100 in persistor.py: 
```
    @staticmethod
    def _project_prefix(project: Text) -> Text:

        p = project or RasaNLUModelConfig.DEFAULT_PROJECT_NAME
        return '{}___'.format(p)
```

**Comment by careless25** [comment]
@wochinge Addressed your comments

File: docs/persist.rst [hunk 6]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
 
     If there is no container with the name ``AZURE_CONTAINER`` Rasa will create it.
 
-Models are gzipped before saving to cloud.
+Models are gzipped before they are saved in the cloud. The gzipped file naming convention
+is `{PROJECT}___{MODEL_NAME}.tar.gz` and it is stored in the root folder of the storage service.
```
Comment (wochinge) [inline docs/persist.rst]: @careless25 Sure it is not `{PROJECT}{MODEL_NAME}.tar.gz`?

File: docs/persist.rst [hunk 7]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+Currently, you are not able manually to specify the path on cloud storage.
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
Currently, you are not able to manually specify the path on the cloud storage.
```

File: docs/persist.rst [hunk 8]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+
+If storing trained models, Rasa NLU will gzip the new model and upload to the container. If retrieving/loading models
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
If storing trained models, Rasa NLU will gzip the new model and upload it to the container. If retrieving/loading models
```

File: docs/persist.rst [hunk 9]
```diff
@@ -55,7 +55,13 @@ Rasa NLU supports using `S3 <https://aws.amazon.com/s3/>`_ and
...
+from the cloud storage, Rasa NLU will download the gzipped model locally an extract the contents to the location
```
Comment (wochinge) [inline docs/persist.rst]:
```suggestion
from the cloud storage, Rasa NLU will download the gzipped model locally and extract the contents to the location
```

**Comment by wochinge** [comment]
@careless25 Thanks! I added a few more comments!

File: docs/persist.rst (hunk 6 above)
Comment (careless25) [reply docs/persist.rst]: Yes, see lines 96-100 in persistor.py

**Comment by careless25** [comment]
@wochinge Fixed. For the `___` separator see: https://github.com/RasaHQ/rasa_nlu/blob/b45b82fc7918049895dedd1916a7a109c60eb239/rasa_nlu/persistor.py#L100

File: docs/persist.rst (hunk 6 above)
Comment (wochinge) [reply docs/persist.rst]: 👍 You know the code better than I - awesome job!

**Pull request merged**